Storage/Bookmarks/Bookmarks.swift

```diff
@@ -14,6 +14,10 @@ public protocol SearchableBookmarks: class {
     func bookmarksByURL(_ url: URL) -> Deferred<Maybe<Cursor<BookmarkItem>>>
 }
 
+public protocol CountableBookmarks: class {
+    func numberOfBookmarks() -> Deferred<Maybe<Int>>
```

Comment:
@sleroux: it sure beats getting all the bookmarks and counting them!

I'd be specific about what it's counting, though. Is it counting bookmarks created on this device, or anywhere? Only bookmarks in the Mobile Bookmarks folder? Does it include folders, livemarks, separators? Will it double-count bookmarks that haven't been merged yet, or does it do GUID-based or content-based deduping? And so on.

Further, this will be sufficient to uniquely identify some users. What approach are you taking to anonymize these numbers?